app.js

```diff
@@ -0,0 +1,295 @@
+import './style.css';
+import kbdkeys from './keys';
+
+const states = {
+  firedKey: null,
+  ControlLeft: false,
+  AltLeft: false,
+  ShiftLeft: false,
+  position: null,
+  lang: 'en',
+};
+
+const createEl = (tag, cls, addTo, tagvalue) => {
+  const el = document.createElement(tag);
+  if (cls != null) {
+    el.setAttribute('class', cls);
+  }
+  if (tagvalue != null) {
+    el.innerText = tagvalue;
+  }
+
+  addTo.append(el);
+  return el;
+};
+
+const changeLang = (kbdKeysInst) => {
+  const { lang, ShiftLeft } = states;
+  if (lang === 'en' && ShiftLeft === false) {
+    kbdKeysInst.forEach((keyGroup) => {
+      keyGroup.forEach((key) => {
+        const keyCurrent = document.querySelector(`div[data-action=${key[1]}]`);
```

Comment:
You do a lot of request to the DOM, so I think you can save all the keys at the array at the moment of creating and work with that array but not with the DOM directly. Every call of querySelector is time